Jenkinsfile.windows

```
@@ -0,0 +1,281 @@
+                  stage ('Checkout') {
+                    when { expression { params.COMMIT_HASH !=  env.GIT_BRANCH } }
+                    steps {
+                      echo "Commit_hash value: ${params.COMMIT_HASH}"
+                        checkout([$class: 'GitSCM',
+                          branches: [[name: "${params.COMMIT_HASH}"]],
+                          extensions: [],
+                          userRemoteConfigs: [[credentialsId: 'github-rstudio-jenkins', url: 'https://github.com/rstudio/rstudio']]])
+                    }
+                  }
+
+                  stage('Build') {
+
+                    environment {
+                      CODESIGN_KEY = credentials('ide-windows-signing-pfx')
+                      CODESIGN_PASS = credentials('ide-pfx-passphrase')
+                    }
+
+                    steps {
+                      // set requisite environment variables and build rstudio
+                      bat "cd package/win32 &&" +
+                        "set \"rstudio_version_major=${RSTUDIO_VERSION_MAJOR}\" &&" +
+                        "set \"rstudio_version_minor=${RSTUDIO_VERSION_MINOR}\" &&" +
+                        "set \"rstudio_version_patch=${RSTUDIO_VERSION_PATCH}\" &&" +
+                        "set \"rstudio_version_suffix=${RSTUDIO_VERSION_SUFFIX}\" &&" +
+                        "set \"package_os=windows\" &&" +
+                        "make-package.bat clean ${FLAVOR} &&" +
+                        "cd ../.."
+                    }
+                  }
+
+                  stage('Tests') {
+                    steps {
+                      bat 'cd package/win32/build/src/cpp && rstudio-tests.bat --scope core'
+                    }
+                  }
+
+                  stage('Electron Tests') {
+                    when { environment name: 'FLAVOR', value: 'Electron' }
+                    steps {
+                      bat 'cd src/node/desktop && scripts\\run-unit-tests.cmd'
+                    }
+                  }
+
+                  stage('Sign, Upload, and Publish') {
+                    when { expression { return params.PUBLISH } }
+
+                    environment {
+                      PACKAGE_VERSION = "${RSTUDIO_RELEASE}".replace('+', '-')
+                    }
+
+                    stages {
+                      stage('Sign') {
+                        environment {
+                          PFX_FILE = credentials('ide-windows-signing-pfx') 
+                          PFX_PASS = credentials('ide-pfx-passphrase')
+                        }
+
+                        steps {
+                          script {
+                            def packageName = "RStudio-${PACKAGE_VERSION}-RelWithDebInfo"
+
+                              bat "\"C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.19041.0\\x86\\signtool\" sign /f %pfx-file% /p %PFX_FILE% %PFX_PASS% /v /debug /n \"RStudio PBC\" /t http://timestamp.digicert.com  package\\win32\\build\\${packageName}.exe"
```

We need to use single quotes around the PFX_PASS portion of this command to avoid insecure interpolation.
```suggestion
                              bat '"C:\\Program Files (x86)\\Windows Kits\\10\\bin\\10.0.19041.0\\x86\\signtool" sign /f %pfx-file% /p %PFX_FILE% %PFX_PASS% /v /debug /n "RStudio PBC" /t http://timestamp.digicert.com  package\\win32\\build\\' + packageName + '.exe'
```